src/misc/msccl/msccl_lifecycle.cc

```diff
@@ -63,12 +63,7 @@ bool mscclAvailable(int rank) {
   return mscclEnabled() && mscclInitialized(rank);
 }
 
-static bool mscclCommCompatible(ncclComm_t comm) {
-  if (rcclParamMscclEnableSingleProcess()) {
-    // Single process usage enabled. No need to guard against multi-thread.
-    return true;
-  }
-
+static bool mscclIsMultithreadedComm(ncclComm_t comm) {
```

Comment:
I'd suggest a different name for this function, as it doesn't actually return whether it is running multi-threaded.
```suggestion
static bool allProcessHostsUnique(ncclComm_t comm) {
```